bin/src/run/watch.ts

```diff
@@ -156,8 +156,9 @@ export default function watch(
 	process.on('uncaughtException', close);
 
 	// only listen to stdin if it is a pipe
-	if (!process.stdin.isTTY) {
+	if (!process.stdin.isTTY || command.watchStdin) {
 		process.stdin.on('end', close); // in case we ever support stdin!
+		process.stdin.resume();
```

Comment:
This line looks like the only relevant fix?

It's not clear to me why we need `--watch-stdin`. If `rollup` is called directly, `process.stdin` is already a flowing stream, and moreover `Ctrl+C` is handled by `onExit`, not this conditional. What use case does `--watch-stdin` allow?

Reply:
If that's the case, it should be the only change in the PR.